contracts/orbital-core/src/contract.rs

```diff
@@ -1,119 +1,171 @@
+};
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
 use cw2::set_contract_version;
-use cw_ownable::{
-    assert_owner, get_ownership, initialize_owner, update_ownership, Action, OwnershipError,
+use cw_ownable::{get_ownership, initialize_owner};
+use neutron_sdk::{
+    bindings::{msg::NeutronMsg, query::NeutronQuery},
+    sudo::msg::SudoMsg,
+    NeutronResult,
 };
 
 use crate::{
-    account_types::UncheckedOrbitalDomainConfig,
-    error::ContractError,
-    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
-    state::{UserConfig, ORBITAL_DOMAINS, USER_CONFIGS},
+    msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg},
+    state::{CLEARING_ACCOUNTS, ORBITAL_DOMAINS, USER_CONFIGS},
 };
 use cosmwasm_std::{
-    to_json_binary, Addr, Binary, BlockInfo, Deps, DepsMut, Env, MessageInfo, Response, StdResult,
+    to_json_binary, Addr, Binary, Deps, DepsMut, Env, MessageInfo, Reply, Response, StdError,
+    StdResult, Uint64,
 };
 
 pub const CONTRACT_NAME: &str = "orbital-core";
 pub const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+pub type OrbitalResult = NeutronResult<Response<NeutronMsg>>;
+pub type QueryDeps<'a> = Deps<'a, NeutronQuery>;
+pub type ExecuteDeps<'a> = DepsMut<'a, NeutronQuery>;
+
 #[entry_point]
 pub fn instantiate(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     _env: Env,
     _info: MessageInfo,
     msg: InstantiateMsg,
-) -> Result<Response, OwnershipError> {
+) -> OrbitalResult {
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
     initialize_owner(deps.storage, deps.api, Some(&msg.owner))?;
 
+    USER_NONCE.save(deps.storage, &Uint64::zero())?;
     Ok(Response::new())
 }
 
 #[entry_point]
 pub fn execute(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     env: Env,
     info: MessageInfo,
     msg: ExecuteMsg,
-) -> Result<Response, ContractError> {
+) -> NeutronResult<Response<NeutronMsg>> {
     match msg {
+        // admin action to manage ownership of orbital-core
         ExecuteMsg::UpdateOwnership(action) => {
-            admin_update_ownership(deps, &env.block, &info.sender, action)
+            admin::transfer_admin(deps, &env.block, &info.sender, action)
         }
+        // admin action to enable new domain for user registration
         ExecuteMsg::RegisterNewDomain {
             domain,
             account_type,
-        } => admin_register_new_domain(deps, info, domain, account_type),
-        ExecuteMsg::RegisterUser {} => register_user(deps, env, info),
+        } => admin::register_new_domain(deps, info, domain, account_type),
+        // user action to create a new user account which enables registration to domains
+        ExecuteMsg::RegisterUser {} => user::register(deps, env, info),
+        // user action to register a new domain which creates their clearing account
+        ExecuteMsg::RegisterUserDomain { domain } => {
+            user::register_new_domain(deps, env, info, domain)
+        }
```

Comment:
hmm, I was thinking to do those default domain registration calls from the front-end. these domain registration calls cost extra fees, and by bundling multiple domain registration messages into a single one it would make asserting the registration fees and handling the callbacks more complicated?